src/jit/MemoryInl.h

```diff
@@ -439,6 +539,24 @@ static void emitLoad(sljit_compiler* compiler, Instruction* instr)
 #endif /* HAS_SIMD */
 
 #if (defined SLJIT_32BIT_ARCHITECTURE && SLJIT_32BIT_ARCHITECTURE)
+#if defined(ENABLE_EXTENDED_FEATURES)
+    if (isAtomic && opcode == SLJIT_MOV) {
```

Comment:
If you add a helper function, which is called directly after `I64AtomicLoadOpcode`, then you don't need isAtomic boolean.